source/Test.cpp

```diff
@@ -482,16 +474,7 @@ void Test::Step(Context &context, UI &menuPanels, UI &gamePanels, PlayerInfo &pl
 					// TODO: handle keys also in-flight (as single inputset)
 					// TODO: combine keys with mouse-inputs
 					for(const string &key : stepToRun.inputKeys)
-					{
-						const char* inputChar = key.c_str();
-						if(!menuPanels.IsEmpty())
-						{
-							if(!KeyInputToUI(menuPanels, inputChar, stepToRun.modKeys))
-								Fail(context, player, "key input on menuPanel failed");
-						}
-						else if(!KeyInputToUI(gamePanels, inputChar, stepToRun.modKeys))
-							Fail(context, player, "key input on gamePanel failed");
-					}
+						context.inputEvents.emplace(KeyInputToEvent(key.c_str(), stepToRun.modKeys));
```

Comment:
Why not call [`SDL_PushEvent`](https://wiki.libsdl.org/SDL_PushEvent)?

Reply:
I guess I was a little bit sleepy yesterday. The code now uses `SDL_PushEvent`. :)